runtime/atlasmap-maven-plugin/src/main/java/io/atlasmap/maven/GenerateInspectionsMojo.java

```diff
@@ -182,15 +182,12 @@ private void generateInspection(List<String> artifacts, Collection<String> class
             JavaClass c = null;
             // Not even this plugin will be available on this new URLClassLoader
             try (URLClassLoader loader = new URLClassLoader(urls.toArray(new URL[urls.size()]), origTccl)) {
-                Thread.currentThread().setContextClassLoader(loader);
                 clazz = loader.loadClass(className);
                 ClassInspectionService classInspectionService = new ClassInspectionService();
                 classInspectionService.setConversionService(DefaultAtlasConversionService.getInstance());
-                c = classInspectionService.inspectClass(clazz);
+                c = classInspectionService.inspectClass(loader, clazz);
             } catch (ClassNotFoundException | IOException e) {
                 throw new MojoExecutionException(e.getMessage(), e);
-            } finally {
-                Thread.currentThread().setContextClassLoader(origTccl);
```

Comment:
Doesn't seem like you need this finally block since you never set the TCCL to anything before this anymore (i.e., since you deleted line 185)

Reply:
Yep, so I'm removing it. So I'm seeing L191-L192 is also red. Do you see something different?